samcli/commands/exceptions.py

```diff
@@ -25,6 +28,42 @@ def __init__(self, message, wrapped_from=None):
         click.ClickException.__init__(self, message)
 
 
+class UnhandledException(click.ClickException):
+    """
+    Exception class to re-wrap any exception that is not a UserException.
+    Typically this means there is a bug in SAM CLI.
+    """
+
+    GH_BUG_REPORT_URL = "https://github.com/aws/aws-sam-cli/issues/new?template=Bug_report.md&title={title}"
+    # NOTE (hawflau): actual exitcode is 1 to not break existing behavior. Only report 255 to telemetry
+    exit_code = 1
+
+    def __init__(self, command: str, exception: Exception) -> None:
+        self._command = command
+        self._exception = exception
+        self.__traceback__ = self._exception.__traceback__
+
+        click.ClickException.__init__(self, type(exception).__name__)
+
+    def show(self, file: t.Optional[t.IO] = None) -> None:
+        """Overriding show to customize printing stack trace and message"""
+        if file is None:
+            file = click._compat.get_text_stderr()  # pylint: disable=protected-access
+
+        tb = "".join(traceback.format_tb(self.__traceback__))
+        click.echo(f"\nTraceback:\n{tb}", file=file, err=True)
+
+        url = self.GH_BUG_REPORT_URL.format(
+            title=urllib.parse.quote(f"{self._command} - {type(self._exception).__name__}")
+        )
+        msg = (
+            f'An unexpected error was encountered while executing "{self._command}".\n'
+            "To create a bug report, follow the Github issue template below:\n"
+            f"{url}"
+        )
```

Comment:
I feel like this could make it easier to have duplicate issues (multiple issues asking about the same problem). Could we first give them a link to search issues with the same title that this issue would be auto-populated?
For example, for the example in your PR description it would be:

https://github.com/aws/aws-sam-cli/issues?q=is%3Aissue+is%3Aopen+samdev+build+-+UnsupportedBuilderException

Reply:
Updated the messaging